src/posh-git.psm1

```diff
@@ -82,18 +81,25 @@ $GitPromptScriptBlock = {
         $prompt += Write-VcsStatus
     }
 
-    # If stopped in the debugger, the prompt needs to indicate that in some fashion
-    $debugMode = (Test-Path Variable:/PSDebugContext) -or [runspace]::DefaultRunspace.Debugger.InBreakpoint
-    if ($debugMode) {
+    # Write default prompt middle if not empty.
+    if ($settings.DefaultPromptMiddle.Text) {
+        $promptMiddle = [PoshGitTextSpan]::new($settings.DefaultPromptMiddle)
+        $promptMiddle.Text = $ExecutionContext.SessionState.InvokeCommand.ExpandString($promptMiddle.Text)
```

Comment:
That is acting kind of like a ctor.  Maybe we could add a parameter to the ctor e.g.:
```posh
$prompt += Write-Prompt [PoshGitTextSpan]::new($settings.DefaultPromptMiddle, ExecutionContext)
```
Or perhaps even better (shorter to type anyway):
```posh
$prompt += Write-Prompt $settings.DefaultPromptMiddle.Expand($ExecutionContext)
```
The `Expand()` method is sort of a fluent API that returns a copy of the original textspan but with the text expanded.  Note: we have to pass in the `$ExecutionContext` or technically the `$ExecutionContext.SessionState.InvokeCommand` object but the former is less to type.  :-)  We have to do this because the context isn't available inside an instance of a PowerShell class.

Reply:
I like the "extension method" approach the best.

>  Note: we have to pass in the $ExecutionContext…because the context isn't available inside an instance of a PowerShell class.

TIL

Reply:
Cool. BTW poked around and found a way to get the $ExecutionContext inside the `Expand()` method so we won't have to pass that in after all.  Turns out PS classes can't find this "automatic" variable when it is compiled but you can work-around it by asking for the variable at runtime with `Get-Variable ExecutionContext -ValueOnly`.  So we should be able to simplify it to just:
```posh
$prompt += Write-Prompt $settings.DefaultPromptMiddle.Expand()
```

Reply:
I've got a fix for both the `DefaultPromptBeforeSuffix` issue and `Expand` but with the branch updates on the server, I can no longer push my changes.  Can we merge this then I'll submit another PR with those two changes?

Reply:
Assuming `origin` is your remote for this repo, and you have two new local commits since I rebased…

```
git fetch
git rebase HEAD~2 --onto origin/rkeithhill/change-debugprompt
```